salt/modules/pdbedit.py

```diff
@@ -92,29 +92,34 @@ def list_users(verbose=True, hashes=False):
 
         if res['retcode'] > 0:
             log.error(res['stderr'] if 'stderr' in res else res['stdout'])
-        else:
-            user_data = {}
-            for user in res['stdout'].splitlines():
-                if user.startswith('-'):
-                    if len(user_data) > 0:
-                        users[user_data['unix username']] = user_data
-                    user_data = {}
-                else:
-                    label = user[:user.index(':')].strip().lower()
-                    data = user[(user.index(':')+1):].strip()
-                    user_data[label] = data
-
-            if len(user_data) > 0:
-                users[user_data['unix username']] = user_data
+            return users
+
+        user_data = {}
+        for user in res['stdout'].splitlines():
+            if user.startswith('-'):
+                if 'unix username' in user_data:
+                    users[user_data['unix username']] = user_data
+                user_data = {}
+            elif ':' in user:
+                label = user[:user.index(':')].strip().lower()
+                data = user[(user.index(':')+1):].strip()
+                user_data[label] = data
+
+        if user_data:
+            users[user_data['unix username']] = user_data
```

Comment:
This probably also applies to what we currently have in develop branch too...

We should probably also check if we actually have ‘unix username’ instead of assuming we do. (See line 100)